src/basic/env-util.c

```diff
@@ -768,6 +768,20 @@ int set_unset_env(const char *name, const char *value, bool overwrite) {
         return 0;
 }
 
+int putenv_dup(const char *assignment, bool override) {
+        const char *e, *n;
+
+        e = strchr(assignment, '=');
+        if (!e)
+                return -EINVAL;
+
+        n = strndupa(assignment, e - assignment);
+
+        if (setenv(n, e + 1, override) < 0)
+                return -errno;
```

Comment:
a quick comment might be nice, along the lines of `/* this is like putenv(), but uses setenv() so that our memory doesn't become part of environ[] */` or so